fastchat/model/model_adapter.py

```diff
@@ -97,9 +97,25 @@ def register_model_adapter(cls):
 @cache
 def get_model_adapter(model_path: str) -> BaseModelAdapter:
     """Get a model adapter for a model_path."""
+    model_last_folder_path = os.path.basename(os.path.normpath(model_path))
+
+    adapter_matched_full_path = None
+    adapter_matched_last_path = None
+
     for adapter in model_adapters:
-        if adapter.match(model_path):
-            return adapter
+        if adapter.match(model_last_folder_path) and type(adapter) != BaseModelAdapter:
+            adapter_matched_last_path = adapter
+        elif adapter.match(model_path) and not adapter_matched_full_path:
+            adapter_matched_full_path = adapter
+        if adapter_matched_last_path and adapter_matched_full_path:
+            break
+
+    if adapter_matched_full_path or adapter_matched_last_path:
+        return (
+            adapter_matched_last_path
+            if adapter_matched_last_path
+            else adapter_matched_full_path
+        )
```

Comment:
The logic is over-complicated. Could you try

```python
    basename = os.path.basename(os.path.normpath(model_path))

    # Try the basename first
    for adapter in model_adapters:
        if adapter.match(basename) and type(adapter) != BaseModelAdapter:
            return adapter

    # Then try the full path
    for adapter in model_adapters:
        if adapter.match(model_path):
            return adapter
```

Reply:
I was hoping to resolve the issue in one loop. It does much simpler by multi loops. :)